solver/src/settlement/settlement_encoder.rs

```diff
@@ -214,14 +225,51 @@ impl SettlementEncoder {
             ],
         }
     }
+
+    // Merge other into self so that the result contains both settlements.
+    // Fails if the settlements cannot be merged for example because the same limit order is used in
+    // both or the same token has different clearing prices.
+    pub fn merge(mut self, mut other: Self) -> Result<Self> {
+        for (key, value) in other.clearing_prices {
+            match self.clearing_prices.entry(key) {
+                Entry::Occupied(entry) => ensure!(*entry.get() == value, "different price"),
+                Entry::Vacant(entry) => {
+                    entry.insert(value);
+                    self.tokens.push(key);
+                }
+            }
+        }
+
+        for other_trade in other.trades.iter() {
+            ensure!(
+                self.trades
+                    .iter()
+                    .all(|self_trade| self_trade.order.order_meta_data.uid
+                        != other_trade.order.order_meta_data.uid),
+                "duplicate trade"
+            );
+        }
+        self.trades.append(&mut other.trades);
+        self.sort_tokens_and_update_indices();
+
+        self.execution_plan.append(&mut other.execution_plan);
+
+        for unwrap in other.unwraps {
+            self.add_unwrap(unwrap);
+        }
+
+        Ok(self)
```

Comment:
Not sure how much logic is in these helper methods, but maybe it would be safer here to create a `SettlementEncoder` and set all fields manually. This way the compiler would yell at us if we create a new field and forget to update the merging logic?

Reply:
Feels weird to destructure self and then restructure just for that. I will add a comment to the field declarations noting `merge`.